webrob/neems/manager.py

```diff
@@ -41,5 +57,13 @@ def filter_neems(self, query_string):
         #   - can an array be added to text index?
         #   - regex search possible, but only without index!!
         #       db.meta.find({"keywords": {"$regex": ".*robot.*","$options": 'i'}},{_id: 1}).pretty()
-        return map(lambda doc: doc["_id"], mongoDBMetaCollection.find(
-            {"$text": {"$search": query_string}}, {"_id": 1}))
+        try:
+            mongoDBMetaCollection = get_mongo_db_meta_collection()
+            return map(lambda doc: doc["_id"], mongoDBMetaCollection.find(
+                {"$text": {"$search": query_string}}, {"_id": 1}))
+
+        except ConnectionFailure as e:
```

Comment:
why not adding a flask handler for this error?

Reply:
done. Now ConnectionFailure has its own flask handler.